app/models/forms/attribute_reset_collection.rb

```diff
@@ -0,0 +1,23 @@
+module Forms
+  class AttributeResetCollection
+    ResetData = Struct.new(:attributes_to_reset, :master_attribute, :enabled_value)
+
+    def initialize
+      @collection = []
+    end
+
+    def add(*args)
+      @collection << ResetData.new(*args)
+    end
+
+    def perform(form)
+      @collection.each do |reset_obj|
+        next if form.public_send(reset_obj.master_attribute) == reset_obj.enabled_value
+
+        reset_obj.attributes_to_reset.each do |attribute|
+          form.public_send("#{attribute}=", nil)
```

Comment:
how about the default value is passed in to the `ResetData` struct by the form itself?
`ResetData = Struct.new(:attributes_to_reset, :master_attribute, :enabled_value, :default_value)`

```
concerning :ResetAttributes do
  included do
    class << self
      def reset(attributes:, if_falsey:, enabled_value: TOGGLE_YES)
        default = model.column_defaults[attribute]
        resettable_attributes.add(attributes, if_falsey, enabled_value, default)
      end
    end
  end
end
```

That would mean the `AttributeResetCollection` didn't need to know so much about the form's API